File: app/src/main/java/com/example/myapplication/Helpers/ContentLoader.java
package com.example.myapplication.Helpers;

import android.app.Activity;

import android.content.Intent;
import android.os.AsyncTask;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentTransaction;

import com.example.myapplication.R;

public class ContentLoader {

    /**
     * Used to loads fragments
     * */


    public static boolean loadActivity(Activity activity1, Activity activity2)
    {
        new ActivityLoader().execute(activity1, activity2);
        return true;
    }

    public static boolean loadFragment(Fragment fragment, Activity activity)
    {
        if(fragment != null)
        {
            FragmentActivity fragmentActivity = (FragmentActivity) activity;
            fragmentActivity.getSupportFragmentManager().beginTransaction().setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE).replace(R.id.fragment_container,fragment).commit();
            return true;
        }
        return false;
    }

    /** TODO Make this class jump between activities **/
    private static class ActivityLoader extends AsyncTask<Activity, Void, Activity> {

        Intent i;
        @Override
        protected Activity doInBackground(Activity... activities) {

            try {
                Thread.sleep(500);
            } catch (InterruptedException e) {
                Thread.interrupted();
            }

            i = new Intent(activities[0], activities[1].getClass());
            i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
            return activities[0];
        }

        @Override
        protected void onPostExecute(Activity activity) {
            activity.startActivity(i);
        }

        @Override
        protected void onPreExecute() {}

        @Override
        protected void onProgressUpdate(Void... values) {}
    }
    /** TODO Make this class jump between activities **/

}

File: app/src/main/java/com/example/myapplication/Helpers/ContentLoader.kt
package com.example.myapplication.Helpers

import android.app.Activity
import android.content.Intent
import android.os.AsyncTask
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentTransaction
import com.example.myapplication.R

object ContentLoader {
    /**
     * Used to loads fragments
     */
    fun loadActivity(activity1: Activity?, activity2: Activity?): Boolean {
        ActivityLoader().execute(activity1, activity2)
        return true
    }

    fun loadFragment(fragment: Fragment?, activity: Activity): Boolean {
        if (fragment != null) {
            val fragmentActivity = activity as FragmentActivity
            fragmentActivity.supportFragmentManager.beginTransaction().setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE).replace(R.id.fragment_container, fragment).commit()
            return true
        }
        return false
    }

    /** TODO Make this class jump between activities  */
    private class ActivityLoader : AsyncTask<Activity?, Void?, Activity>() {
        var i: Intent? = null
        protected override fun doInBackground(vararg activities: Activity): Activity {
            try {
                Thread.sleep(500)
            } catch (e: InterruptedException) {
                Thread.interrupted()
            }
            i = Intent(activities[0], activities[1].javaClass)
            i!!.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            return activities[0]
        }

        override fun onPostExecute(activity: Activity) {
            activity.startActivity(i)
        }

        override fun onPreExecute() {}
        protected override fun onProgressUpdate(vararg values: Void) {}
    }
    /** TODO Make this class jump between activities  */
}